CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+# Contributing
+
+When contributing to this repository, please first discuss the change you wish
+to make via issue, email, or any other method with the owners of this
+repository before making a change.
+
+Please note we have a code of conduct, please follow it in all your
+interactions with the project.
+
+## Pull Request Process
+
+1. Ensure any install or build dependencies are removed before the end of the
+   layer when doing a build.
+2. Update the `README.md` and `CHANGELOG.md` with details of changes to the
+   interface, this includes new environment variables, exposed ports, useful
+   file locations and container parameters.
+3. Increase the version numbers in any examples files and the `README.md` to
+   the new version that this Pull Request would represent. The versioning
+   scheme we use is [SemVer](http://semver.org/).
+4. You may merge the Pull Request in once you have the sign-off of two other
+   developers, or if you do not have permission to do that, you may request the
+   second reviewer to merge it for you.
```

Comment:
I think the process we want to promote instead is:

1. Fork repo
2. Branch off master branch and do fix
3. PR to merge into master upstream
4. We review and if we approve we merge, or we give comments.

So the audience for this file is not members of the repo that have write access like you and I but people outside of OSC that would like to make small contributions like https://github.com/OSC/ood-shell/pull/34

Reply:
We might also want to mention that the README of the main repo will provide links to the components, the apps, the documentation repo, and the website. And that the process for making a contribution to the documentation repo and the website is different from the process of making a contribution to anything else.

Finally, the components we advertise on the main README are missing things for developers, like the documentation and test documentation repos, the rpm repos, etc. so we may want to add that to the README for developers and then link to this contributing doc from the README.